vertical-pod-autoscaler/pkg/utils/vpa/capping.go

```diff
@@ -27,47 +27,53 @@ import (
 func GetCappedRecommendationForContainer(
 	container apiv1.Container,
 	podRecommendation *vpa_types.RecommendedPodResources,
-	policy *vpa_types.PodResourcePolicy) (apiv1.ResourceList, error) {
+	policy *vpa_types.PodResourcePolicy) (*vpa_types.RecommendedContainerResources, error) {
 	containerRecommendation := getRecommendationForContainer(podRecommendation, container)
 	if containerRecommendation == nil {
 		return nil, fmt.Errorf("no recommendation available for container name %v", container.Name)
 	}
 	// containerPolicy can be nil (user does not have to configure it).
 	containerPolicy := getContainerPolicy(container.Name, policy)
-	applyVPAPolicy(containerRecommendation, containerPolicy)
-	// TODO: If limits and policy are conflicting, set some condition on the VPA.
-	capRecommendationToContainerLimit(containerRecommendation, container)
-	res := make(apiv1.ResourceList)
-	for resource, recommended := range containerRecommendation.Target {
-		res[resource] = recommended
+
+	cappedRecommendations := containerRecommendation.DeepCopy()
```

Comment:
Why DeepCopy is needed here?

Reply:
Well, previously it was also a "deep copy" of containerRecommendation.Target (just written explicitly, copying field by field).
As we now want to cap other fields (MinRecommended and MaxRecommended) as well, it would be very verbose to explicitly copy every field. Instead I'm just calling DeepCopy() which does it for me.

Reply:
Oh, you're probably right that the copy wasn't necessary in the first place, because getRecommendationForContainer() already returns a copy. But I think I'd prefer not to depend on it, because it's fragile and easy to break without noticing.